runtime/doc/lsp.txt

```diff
@@ -169,73 +170,130 @@ Requests and notifications defined by the LSP specification are referred to as
-        - {err}     (table|nil) Error info dict, or `nil` if the request
-                    completed.
-        - {result}  (Result | Params | nil) `result` key of the |lsp-response| or
-                    `nil` if the request failed.
-        - {ctx}     (table) Table of calling state associated with the
-                    handler, with these keys:
-                    - {method}  (string) |lsp-method| name.
-                    - {client_id} (number) |vim.lsp.client| identifier.
-                    - {bufnr}   (Buffer) Buffer handle.
-                    - {params}  (table|nil) Request parameters table.
-                    - {version} (number) Document version at time of
+        - {err}     (`lsp.LspResponseError?`) Error info table, or `nil` if
+                    the request completed without errors.
+        - {result}  (`Result`) readout of `result` key in the |lsp-response|,
+                    or `nil` if the request failed (i.e. {err} `~= nil`); or
+          {request} (`Params`) readout of `params` key in the |lsp-request| or
+                    |lsp-notification|, or `nil` if the method does not
+                    require params.
+        - {ctx}     (`lsp.HandlerContext`) table of calling state associated
+                    with the handler, with these keys:
+                    - {method}  (`string`) |lsp-method| name.
+                    - {client_id} (`number`) |vim.lsp.client| identifier.
+                    - {bufnr}   (`number`) Buffer handle.
+                    - {params}  (`table?`) Request parameters table.
+                    - {version} (`number`) Document version at time of
                                 request. Handlers can compare this to the
                                 current document version to check if the
                                 response is "stale". See also |b:changedtick|.
-        - {config}  (table) Handler-defined configuration table, which allows
+        - {config}  (`table`) Handler-defined configuration table, which allows
                     users to customize handler behavior.
                     For an example, see:
                         |vim.lsp.diagnostic.on_publish_diagnostics()|
                     To configure a particular |lsp-handler|, see:
-                        |lsp-handler-configuration|
+                        |lsp-handler-configuration| |vim.lsp.with()|
 
-    Returns: ~
-        Two values `result, err` where `err` is shaped like an RPC error: >
+    Return (multiple): ~
+        (`Result?`) `result` on successful request, or `nil` otherwise.
+                         See |lsp-handler-return|
+        (`lsp.ResponseError?`) `error` on unsuccessful request, or `nil` otherwise.
+<
+                                                          *lsp-handler-return*
+Return values and Error handling of |lsp-handler|:
+
+    For |lsp-response| and |lsp-notification| handlers, there are no return values
+    (equivalent as returning `nil`, `nil`).
```

Comment:
> (equivalent as returning `nil`, `nil`)

??

Reply:
Is the meaning unclear? What I menat is: in Lua, `local x, y = (function() end)()` assigns `nil` to both `x` and `y`.

Reply:
OHHHH I got it. The way I read that is "equivalent as returning `nil` (end of sentence here), `nil`".

Maybe "equivalent as returning `nil, nil`" would make it clearer that both `nil`s are being returned?